src/Workspaces/Core/MSBuild/MSBuild/CSharp/CSharpProjectFile.cs

```diff
@@ -0,0 +1,440 @@
+                }
+
+                if (platform != null && target != null)
+                {
+                    break;
+                }
+            }
+
+            if (string.Equals("anycpu32bitpreferred", platform, StringComparison.OrdinalIgnoreCase)
+                && (string.Equals("library", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("module", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("winmdobj", target, StringComparison.OrdinalIgnoreCase)))
+            {
+                return commandLineArgs.SetItem(platformIndex, "/platform:anycpu");
+            }
+
+            return commandLineArgs;
+        }
+
+        private ImmutableArray<string> GetAliases(MSB.Framework.ITaskItem item)
+        {
+            var aliasesText = item.GetMetadata("Aliases");
+
+            if (string.IsNullOrEmpty(aliasesText))
+            {
+                return ImmutableArray<string>.Empty;
+            }
+
+            return ImmutableArray.CreateRange(aliasesText.Split(new char[] { ' ', ',' }, StringSplitOptions.RemoveEmptyEntries));
+        }
+
+        private ImmutableArray<string> ReadCommandLineArguments(MSB.Execution.ProjectInstance project)
+        {
+            var builder = ImmutableArray.CreateBuilder<string>();
+
+            var additionalFiles = this.GetAdditionalFilesFromModel(project);
+            if (additionalFiles != null)
+            {
+                foreach (var additionalFile in additionalFiles)
+                {
+                    builder.Add("/additionalfile:\"" + this.GetDocumentFilePath(additionalFile) + "\"");
+                }
+            }
+
+            var allowUnsafeBlocks = project.ReadPropertyBool("AllowUnsafeBlocks");
+            if (allowUnsafeBlocks)
+            {
+                builder.Add("/unsafe");
+            }
+
+            var analyzers = this.GetAnalyzerReferencesFromModel(project);
+            if (analyzers != null)
+            {
+                foreach (var analyzer in analyzers)
+                {
+                    builder.Add("/analyzer:\"" + this.GetDocumentFilePath(analyzer) + "\"");
+                }
+            }
+
+            var applicationConfiguration = project.ReadPropertyString("AppConfigForCompiler");
+            if (!string.IsNullOrWhiteSpace(applicationConfiguration))
+            {
+                builder.Add("/appconfig:" + applicationConfiguration);
+            }
+
+            var baseAddress = project.ReadPropertyString("BaseAddress");
+            if (!string.IsNullOrWhiteSpace(baseAddress))
+            {
+                builder.Add("/baseaddress:" + baseAddress);
+            }
+
+            var checkForOverflowUnderflow = project.ReadPropertyBool("CheckForOverflowUnderflow");
+            if (checkForOverflowUnderflow)
+            {
+                builder.Add("/checked");
+            }
+
+            var codePage = project.ReadPropertyInt("CodePage");
```

Comment:
all these string literal seems can be changed to const. and have map between MSBUILD terminology to csc terminology

Reply:
I agree. Will do.

Reply:
I've cleaned up and unified this area quite a bit. I didn't create a map since the logic can vary pretty significantly between various command-line arguments. However, the logic is much simpler and there's less of it now.